### PR TITLE
CON-4432 add certificate receiver

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/CertificateTests/WhenUserLoadsAmendsRecipientPage.cs [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 using SFA.DAS.AdminService.Web.Controllers;
-using SFA.DAS.AdminService.Web.ViewModels;
+using SFA.DAS.AdminService.Web.ViewModels.CertificateAmend;
 
 namespace SFA.DAS.AdminService.Web.Tests.Controllers.CertificateTests
 {
     public class WhenUserLoadsAmendsRecipientPage : CertificateAmendQueryBase
     {
-        private IActionResult _result;
+        /*private IActionResult _result;
```
Comment (cofaulco) [inline src/SFA.DAS.AdminService.Web.Tests/Controllers/CertificateTests/WhenUserLoadsAmendsRecipientPage.cs:11]: is this supposed to be commented out?

File: src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+                .Build());
+
+            fixture.Validate();
+
+            fixture.Verify(nameof(CertificateAddressViewModel.AddressLine1), "NotEmptyValidator", "Enter an address", true);
+        }
+
+        [TestCase(CertificateSendTo.Employer)]
+        [TestCase(CertificateSendTo.Apprentice)]
+        public void WhenCityIsNotEmpty_ErrorIsDisplay(CertificateSendTo sendTo)
```
Comment (cofaulco) [inline src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs:220]: should this be `ErrorIsNotDisplayed`?

File: src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+                .Build());
+
+            fixture.Validate();
+
+            fixture.Verify(nameof(CertificateAddressViewModel.AddressLine1), "NotEmptyValidator", "Enter an address", true);
+        }
+
+        [TestCase(CertificateSendTo.Employer)]
+        [TestCase(CertificateSendTo.Apprentice)]
+        public void WhenAddress1NotIsEmpty_ErrorIsDisplay(CertificateSendTo sendTo)
```
Comment (cofaulco) [inline src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs:185]: should this be `ErrorIsNotDisplayed`?

File: src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+                .Build());
+
+            fixture.Validate();
+
+            fixture.Verify(nameof(CertificateAddressViewModel.AddressLine1), "NotEmptyValidator", "Enter an address", true);
+        }
+
+        [TestCase(CertificateSendTo.Employer)]
+        [TestCase(CertificateSendTo.Apprentice)]
+        public void WhenPostcodeIsNotEmpty_ErrorIsDisplay(CertificateSendTo sendTo)
```
Comment (cofaulco) [inline src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs:254]: should this be `ErrorIsNotDisplayed`?

File: src/SFA.DAS.AdminService.Web/ViewModels/CertificateBaseViewModel.cs [hunk 5]
```diff
@@ -23,6 +23,12 @@ protected void BaseFromCertificate(Certificate cert)
             StandardCode = cert.StandardCode;
             Version = CertificateData.Version;
         }
+
+        public virtual Certificate GetCertificateFromViewModel(Certificate certificate, CertificateData certData)
```
Comment (cofaulco) [inline src/SFA.DAS.AdminService.Web/ViewModels/CertificateBaseViewModel.cs:27]: unused param CertificateData. 
Also, what is the use of this method?

**Review by cofaulco — COMMENTED** [review]
couple of minor comments, otherwise looking good

File: src/SFA.DAS.AdminService.Web/ViewModels/CertificateDelete/CertificateConfirmDeleteViewModel.cs [hunk 6]
```diff
@@ -9,22 +9,22 @@ public class CertificateConfirmDeleteViewModel : CertificateBaseViewModel, ICert
...
-            BaseFromCertificate(cert);
+            base.FromCertificate(cert);
 
             Uln = cert.Uln;
             CertificateReference = cert.CertificateReference;
             StandardCode = cert.StandardCode;
         }
 
-        public Certificate GetCertificateFromViewModel(Certificate certificate, CertificateData data)
+        public override Certificate GetCertificateFromViewModel(Certificate certificate, CertificateData data)
```
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AdminService.Web/ViewModels/CertificateDelete/CertificateConfirmDeleteViewModel.cs:23]: It transfers the properties of the class which is called ...ViewModel into a certificate

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/CertificateTests/WhenUserLoadsAmendsRecipientPage.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AdminService.Web.Tests/Controllers/CertificateTests/WhenUserLoadsAmendsRecipientPage.cs:11]: The controller was removed - the test is now removed.

File: src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs (hunk 2 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs:220]: Fixed

File: src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs (hunk 3 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs:185]: Fixed

File: src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs (hunk 4 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.AdminService.Web.Tests/Validators/CertificateCheckViewModelValidatorTests.cs:254]: Fixed

**Review by cofaulco — APPROVED** [review]

**Pull request merged**